### ISSUE TITLE
Add dstack, vstack, and hstack

### ISSUE THREAD

**Issue by ngoldbaum** [rust-ndarray/ndarray#667]
In the vein of #75 which added `stack`, it would be really nice if there were a way to easily convert e.g. `Vec<Array2<D>>` into an `Array3<D>`. Current I'm doing something like this:

```
let imshape = data[0].shape()
let mut activation: Array3<f64> = Array3::zeros((n_batch, imshape[0], imshape[1]));
for (i, mm) in data.iter().enumerate() {
    activation.slice_mut(s![i, .., ..]).assign(&m);
}
```

It would be nice to not have to do this manually, it would also be clearer to people who are used to numpy (IMO).

I think this is equivalent to relaxing the restriction that `axis` in [`ndarray::stack`](https://docs.rs/ndarray/0.12.1/ndarray/fn.stack.html) has to be an existing axis.

**Comment by jturner314** [comment]
What we call `stack`, NumPy calls [`concatenate`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.concatenate.html), and we don't have an equivalent of NumPy's [`stack`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.stack.html). I actually like NumPy's naming (`concatenate` along existing axis, `stack` along new axis) better than `ndarray`'s.

Any thoughts on renaming the existing `stack` function/macro to `concatenate` or `concat` and adding a `stack` function that stacks along a new axis?

**Comment by LukeMathWalker** [comment]
It sounds good to me - it would also help for issues like #650.

**Comment by bluss** [comment]
@jturner314 Sounds good

**Comment by andrei-papou** [comment]
Seems like @bluss is a bit busy so could one of you guys @LukeMathWalker @jturner314 please take a look at the [PR](https://github.com/rust-ndarray/ndarray/pull/735) changes? Thanks a lot!

**Comment by TheButlah** [comment]
New user of ndarray here - coming from python, i expected `ndarray::stack` to have the same semantics as `np.stack`. I would like to also put a vote in for changing the name to `ndarray::concatenate`.

Also, is there a `ndarray` alternative for `np.stack` now? Trying to find the function I should be using, but not seeing one.

**Comment by ngoldbaum** [comment]
No, that’s all in #735, which appears to have stalled.

**Comment by Luthaf** [comment]
Looks like the name has been changed, but documentation in https://docs.rs/ndarray/0.14.0/ndarray/doc/ndarray_for_numpy_users/index.html still refer to `stack!` as the alternative for `np.concatenate`. Should it be updated?

**Comment by bluss** [comment]
That should already be fixed in master, which is released as 0.15 soon.

The present feature request is closed - it was fixed already in 0.14.
